backtrace-library/src/main/java/backtraceio/library/models/json/BacktraceReport.java

```diff
@@ -249,8 +267,7 @@ public BacktraceData toBacktraceData(Context context, Map<String, Object> client
     }
 
     public BacktraceData toBacktraceData(Context context, Map<String, Object> clientAttributes, boolean isProguardEnabled) {
-        BacktraceData backtraceData = new BacktraceData(context, this, clientAttributes);
-        backtraceData.symbolication = isProguardEnabled ? "proguard" : null;
-        return backtraceData;
+        final String symbolication = isProguardEnabled ? "proguard" : null;
+        return new BacktraceData.Builder(context, this, symbolication, clientAttributes).build();
```

Comment:
I see this pattern a second time. Maybe it's worth to pass `isProguardEnabled` into the builder?